codeguru_profiler_agent/sdk_reporter/sdk_reporter.py

```diff
@@ -67,9 +75,17 @@ def refresh_configuration(self):
             # whole process because the customer may fix this on their side by creating/changing the profiling group.
             # We handle service exceptions like this in boto3
             # see https://boto3.amazonaws.com/v1/documentation/api/latest/guide/error-handling.html
-            if error.response['Error']['Code'] in ['ResourceNotFoundException', 'ValidationException']:
+            if error.response['Error']['Code'] == 'ValidationException':
                 self.agent_config_merger.disable_profiling()
-            self._log_request_failed(operation="configure_agent", exception=error)
+                self._log_request_failed(operation="configure_agent", exception=error)
+
+            if error.response['Error']['Code'] == 'ResourceNotFoundException':
```

Comment:
Question here : What configuration will be used in case of `ResourceNotFoundException` and `ValidationException`.

Reply:
The default configuration defined [here](https://github.com/aws/amazon-codeguru-profiler-python-agent/blob/main/codeguru_profiler_agent/profiler.py#L107)